### PR TITLE
taints float is 2.2 spec.

### PR DESCRIPTION
**Pull request by unak**

### CODE REVIEW

**Comment by eregon** [comment]
Is this due to failures on rubyci.org?
I always wondered why it passes on some architectures but not on others, any ideas?
For instance, locally, using the oldest 2.1 I have I cannot reproduce:
```
$ ruby -ve 'p Marshal.load(Marshal.dump([1.2]).taint)[0].tainted?'
ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-linux]
false
```
Otherwise, if this is indeed spec only in 2.2, I'm fine with this changed but it would be good to know if this is a bug of MRI 2.1 and before.

File: core/marshal/shared/load.rb [hunk 1]
```diff
@@ -227,7 +227,9 @@
       x = [1.2]
       y = Marshal.send(@method, Marshal.dump(x).taint)
       y.tainted?.should be_true
-      y.first.tainted?.should be_false
+      ruby_version_is "2.2" do
+        y.first.tainted?.should be_false
+      end
```
Comment (eregon) [inline core/marshal/shared/load.rb]: the `ruby_version_is` should be around the whole `it` example, as above.

**Comment by unak** [comment]
> Is this due to failures on rubyci.org?
Yes.

> I always wondered why it passes on some architectures but not on others, any ideas?
On 64bit platforms, '1.2' is a Flonum.
Flonum is immediate value, so it can not have taint flag.
After releasing 2.1.0, we decided to change the behavior of old Float values.
Therefore, with 2.2 or later, any Float values can not have taint flag.

In short, this is a spec change at 2.2.

**Comment by eregon** [comment]
I see, thanks for the explanation, let's merge this!

File: core/marshal/shared/load.rb (hunk 1 above)
Comment (eregon) [inline core/marshal/shared/load.rb]: I fixed it in c559cd13cf1e4ca65efc6e3c6560095db638c873.

**Comment by unak** [comment]
Thank you!

**Pull request merged**